test/bigtable/table.js

```diff
@@ -0,0 +1,852 @@
+
+function FakeRow() {
+  this.calledWith_ = arguments;
+  Row.apply(this, arguments);
+}
+
+FakeRow.formatChunks_ = sinon.spy(function(chunks) {
+  return chunks;
+});
+
+nodeutil.inherits(FakeRow, Row);
+
+var FakeMutation = {
+  methods: Mutation.methods,
+  convertToBytes: sinon.spy(function(value) {
+    return value;
+  }),
+  convertFromBytes: sinon.spy(function(value) {
+    return value;
+  }),
+  parse: sinon.spy(function(value) {
+    return value;
+  })
+};
+
+var FakeFilter = {
+  parse: sinon.spy(function(value) {
+    return value;
+  })
+};
+
+describe('Bigtable/Table', function() {
+  var TABLE_ID = 'my-table';
+  var BIGTABLE = {
+    clusterName: 'a/b/c/d'
+  };
+  var TABLE_NAME = BIGTABLE.clusterName + '/tables/' + TABLE_ID;
+
+  var Table;
+  var table;
+
+  before(function() {
+    mockery.registerMock(
+      '../../lib/common/grpc-service-object', FakeGrpcServiceObject);
+    mockery.registerMock('../../lib/bigtable/family', FakeFamily);
+    mockery.registerMock('../../lib/bigtable/mutation', FakeMutation);
+    mockery.registerMock('../../lib/bigtable/filter', FakeFilter);
+    mockery.registerMock('pumpify', pumpify);
+    mockery.registerMock('../../lib/bigtable/row', FakeRow);
+
+    mockery.enable({
+      useCleanCache: true,
+      warnOnUnregistered: false
+    });
+
+    Table = require('../../lib/bigtable/table.js');
+  });
+
+  after(function() {
+    mockery.deregisterAll();
+    mockery.disable();
+  });
+
+  beforeEach(function() {
+    table = new Table(BIGTABLE, TABLE_ID);
+  });
+
+  afterEach(function() {
+    sinon.restore();
+
+    Object.keys(FakeMutation).forEach(function(spy) {
+      if (FakeMutation[spy].reset) {
+        FakeMutation[spy].reset();
+      }
+    });
+
+    FakeFilter.parse.reset();
+  });
+
+  describe('instantiation', function() {
+    it('should inherit from GrpcServiceObject', function() {
+      var config = table.calledWith_[0];
+
+      assert(table instanceof FakeGrpcServiceObject);
+      assert.strictEqual(config.parent, BIGTABLE);
+      assert.strictEqual(config.id, TABLE_NAME);
+
+      assert.deepEqual(config.methods, {
+        create: true,
+        delete: {
+          protoOpts: {
+            service: 'BigtableTableService',
+            method: 'deleteTable'
+          },
+          reqOpts: {
+            name: TABLE_NAME
+          }
+        },
+        exists: true,
+        get: true,
+        getMetadata: {
+          protoOpts: {
+            service: 'BigtableTableService',
+            method: 'getTable'
+          },
+          reqOpts: {
+            name: TABLE_NAME
+          }
+        }
+      });
+    });
+
+    it('should use Bigtable#createTable to create the table', function(done) {
+      var fakeOptions = {};
+
+      BIGTABLE.createTable = function(name, options, callback) {
+        assert.strictEqual(name, TABLE_ID);
+        assert.strictEqual(options, fakeOptions);
+        callback(null, table); // done
```

Comment:
Have to pass it in because `create` is made via GrpcServiceObject config and it's going to try and read metadata from the table. Is there a better way to test this? e.g. calling `createMethod` directly..?